store/RedisStore.js

```diff
@@ -20,13 +26,20 @@ const lua = `
   if ttl == -1 or (continueExceeding and current > max) then
     redis.call('PEXPIRE', key, timeWindow)
     ttl = timeWindow
+  
+  -- If the key is new or if its incremented value has exceeded the max value and exponential backoff is enabled then set its TTL
+  elseif exponentialBackoff and current > max then
+    local backoffExponent = current - max - 1
+    ttl = math.min(timeWindow * (2 ^ backoffExponent), MAX_SAFE_INTEGER)
+    redis.call('PEXPIRE', key, ttl)
```

Comment:
We keep incrementing `current`, so what happens if we surpass the max integer in Lua? It doesn't become Infinity but actually flips to negative

Consider this:

```lua
local max = math.maxinteger
print (max * 2) // -2
```

Then `math.min` will take -2 as `ttl`

The easiest solution is to cast to float, which also makes the number become Infinite once it's out of bounds

Reply:
This is why we should add a Redis test as well

Reply:
I recommend doing this at the top:

```lua
local timeWindow = tonumber(ARGV[1]) * 1.0
```

Reply:
Done